Plugins/ExaTrkX/src/TorchEdgeClassifier.cpp

```diff
@@ -19,9 +20,19 @@ namespace Acts {
 
 TorchEdgeClassifier::TorchEdgeClassifier(const Config& cfg,
                                          std::unique_ptr<const Logger> _logger)
-    : m_logger(std::move(_logger)), m_cfg(cfg) {
+    : m_logger(std::move(_logger)),
+      m_cfg(cfg),
+      m_device(torch::Device(torch::kCPU)) {
   c10::InferenceMode guard(true);
   m_deviceType = torch::cuda::is_available() ? torch::kCUDA : torch::kCPU;
+  if (m_deviceType == torch::kCUDA && cfg.deviceID >= 0 &&
+      static_cast<std::size_t>(cfg.deviceID) < torch::cuda::device_count()) {
+    ACTS_DEBUG("GPU device " << cfg.deviceID << " is being used.");
+    m_device = torch::Device(torch::kCUDA, cfg.deviceID);
+  } else {
+    ACTS_ERROR("GPU device " << cfg.deviceID
+                             << " not available. Using CPU instead.");
```

Comment:
"Using CPU instead" seem to imply that you are going to continue running. In that case this should maybe be a warning and not an error ?

Reply:
Fixed
https://github.com/hrzhao76/acts/commit/b6e52ef8c12911e13b1761539b9999e179b3a92e#diff-ff0f8240ceba0d54ffd3367c34b599d8cdec8bafa206561b428e20dcb50d7498R33